### PR TITLE
Create generic move documents for HHG+PPM moves [delivers #164085600]

### PR DESCRIPTION
**Pull request by jim**
## Description

This PR fixes an issue where a TSP user couldn't upload documents to an HHG+PPM move due to code that hadn't been updated to handle that move type.

## Reviewer Notes

I added another argument to `testdatagen.CreateShipmentOfferData`, which I don't love, but it was the best thing I could come up with other than rewriting that function altogether.

## Setup

1. Setup your local database:

    ```sh
    $ make db_dev_e2e_populate
    ```

2. Run `make server_run` and `make tsp_client_run`.

3. Login as `tspuser1@example.com`.

4. Double-click on the move with locator `HHGPPM` under `New Moves`.

5. Click on `Accept Move`.

6. Click `Upload new document` to open the document viewer in another tab.

7. Fill out the fields (you can use the 'other' document type) and select a file.

8. Click `Save`. The document should appear in the right column document list.

9. Close that tab and return to the move detail screen.

10. Reload the page and verify that the document you just added appears in the sidebar.

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](./docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](./docs/backend.md#querying-the-database-safely)
 have been satisfied.

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/164085600) for this change
* [Zendesk ticket](https://mymovemil.zendesk.com/tickets/184)

### CODE REVIEW

File: pkg/testdatagen/make_shipment_offer_records.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func MakeDefaultShipmentOffer(db *pop.Connection) models.ShipmentOffer {
 // CreateShipmentOfferData creates a list of TSP Users, Shipments, and Shipment Offers
 // Must pass in the number of tsp users to create and number of shipments.
 // The split of shipment offers should be the length of TSP users and the sum should equal the number of shipments
-func CreateShipmentOfferData(db *pop.Connection, numTspUsers int, numShipments int, numShipmentOfferSplit []int, statuses []models.ShipmentStatus) ([]models.TspUser, []models.Shipment, []models.ShipmentOffer, error) {
+func CreateShipmentOfferData(db *pop.Connection, numTspUsers int, numShipments int, numShipmentOfferSplit []int, statuses []models.ShipmentStatus, moveType models.SelectedMoveType) ([]models.TspUser, []models.Shipment, []models.ShipmentOffer, error) {
```
Comment (stangah) [inline pkg/testdatagen/make_shipment_offer_records.go]: Is moveType being used anywhere? There's a `selectedMoveType` being used in this function too

File: pkg/handlers/publicapi/generic_move_documents_test.go [hunk 2]
```diff
@@ -76,15 +75,31 @@ func (suite *HandlerSuite) TestCreateGenericMoveDocumentHandler() {
...
-	badUserResponse := handler.Handle(newMoveDocParams)
-	suite.CheckResponseForbidden(badUserResponse)
+	//badUserResponse := handler.Handle(newMoveDocParams)
+	//suite.CheckResponseForbidden(badUserResponse)
 
 	// Now try a bad shipment
-	newMoveDocParams.ShipmentID = strfmt.UUID(uuid.Must(uuid.NewV4()).String())
-	badMoveResponse := handler.Handle(newMoveDocParams)
-	suite.CheckResponseForbidden(badMoveResponse)
+	//newMoveDocParams.ShipmentID = strfmt.UUID(uuid.Must(uuid.NewV4()).String())
```
Comment (stangah) [inline pkg/handlers/publicapi/generic_move_documents_test.go]: Can this be removed?

File: pkg/testdatagen/make_shipment_offer_records.go (hunk 1 above)
Comment (jim) [reply pkg/testdatagen/make_shipment_offer_records.go]: I dunno what happened here. The argument is the one we want; the other was intended to be deleted. Will fix.

File: pkg/handlers/publicapi/generic_move_documents_test.go (hunk 2 above)
Comment (jim) [reply pkg/handlers/publicapi/generic_move_documents_test.go]: I put it back. Didn't mean to leave it commented out.

**Pull request merged**